### PR TITLE
CorrDiff: CWB Dataloader cleanups

### PR DESCRIPTION
**Pull request by nbren12**
# Modulus Pull Request

## Description

This PR deletes some unused code that made the CWB data loader harder to follow. It also uncomments, the corrdiff dataset tests.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: examples/generative/corrdiff/datasets/cwb.py [hunk 1]
```diff
@@ -347,97 +254,46 @@ def __init__(
...
-        n_history=0,
-        min_path=None,
-        max_path=None,
-        global_means_path=None,
-        global_stds_path=None,
-        normalization="v1",
     ):
+
+        self._dataset = dataset
         if not all_times:
```
Comment (nbren12) [inline examples/generative/corrdiff/datasets/cwb.py:264]: please review this carefully for proper train test split.

**Comment by nbren12** [comment]
#589

**Comment by mnabian** [comment]
Need to merge this first for the CI to pass: https://github.com/NVIDIA/modulus/pull/594

File: examples/generative/corrdiff/tests/test_dataset.py [hunk 2]
```diff
@@ -14,55 +14,69 @@
...
-#     assert dataset.output_channels()
+def test_zarr_dataset_channels(dataset):
+    assert dataset.input_channels()
+    assert dataset.output_channels()
 
 
-# def test_zarr_dataset_time(dataset):
-#     isinstance(dataset.time()[0].year, int)
+def test_zarr_dataset_time(dataset):
+    isinstance(dataset.time()[0].year, int)
```
Comment (mnabian) [inline examples/generative/corrdiff/tests/test_dataset.py:55]: `assert` is missing?

File: examples/generative/corrdiff/tests/test_dataset.py [hunk 3]
```diff
@@ -14,55 +14,69 @@
...
+    assert isinstance(ans, np.int64)
 
-# def test_filter_time():
-#     class MockData(torch.utils.data.Dataset):
 
-#         def __getitem__(self, idx):
-#             return self.time()[idx]
+def test_train_test_split():
+    ds = get_zarr_dataset(data_path=path, train=True)
+    assert not any(t.year == 2021 for t in ds.time())
```
Comment (mnabian) [inline examples/generative/corrdiff/tests/test_dataset.py:65]: I'm not sure about this test... What if we want to train on 2021?

File: examples/generative/corrdiff/datasets/cwb.py [hunk 4]
```diff
@@ -347,97 +254,46 @@ def __init__(
...
-            self._dataset = (
-                FilterTime(dataset, is_not_2021)
+            self._time_indices = (
+                [i for i, t in enumerate(self._dataset.time()) if t.year != 2021]
```
Comment (mnabian) [inline examples/generative/corrdiff/datasets/cwb.py:266]: I wish we could avoid hard-coding the year here

**Review by mnabian — APPROVED** [review]

**Comment by jleinonen** [comment]
Pretty big changes here, so can you check that you can start training ok, and that inference results with the pretrained checkpoint look ok?

File: examples/generative/corrdiff/tests/test_dataset.py (hunk 3 above)
Comment (nbren12) [reply examples/generative/corrdiff/tests/test_dataset.py:65]: The 2021 hardcodes already existed so not new in the PR. I just moved the harcodes to new places. Would prefer to not remove rn, but could parameterize it in a future PR.

**Comment by nbren12** [comment]
> Pretty big changes here, so can you check that you can start training ok, and that inference results with the pretrained checkpoint look ok?

Hey @Jussmith01. I'm not that spun-up on corrdiff training/inference, but did add a regression test against the real data, and ensured that the value returned was bit-for-bit identical throughout the refactor. Perhaps you can add a simple end-to-end test that runs the training and inference on a desktop. Until such is in the code base, I think it's a bit laborious to make it a requirement for merging. Thoughts @mnabian?

File: examples/generative/corrdiff/datasets/cwb.py (hunk 4 above)
Comment (nbren12) [reply examples/generative/corrdiff/datasets/cwb.py:266]: was hardcoded before, so this would be a "new feature". Would prefer to keep the new feature code separate from the refactors in this PR.

**Comment by nbren12** [comment]
Ready for re-review. I decided to leave the old reference configs intact since they have the hardcoded validation and other times. We'll probably want to include that in some form later on.

**Comment by nbren12** [comment]
@jleinonen. `normalize_input` and `normalize_output` are covered by my existing regression check (since they are called in `__getitem__`). I will add a test for the denormalize methods.

**Review by jleinonen — APPROVED** [review]
